scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -582,7 +583,10 @@ impl TomlManifest {
             .clone()
             .map(|edition| edition.resolve("edition", || inheritable_package.edition()))
             .transpose()?
-            .unwrap_or_default();
+            .unwrap_or_else(|| {
+                config.ui().warn("`edition` not found in 'package' section");
```

Comment:
```suggestion
                config.ui().warn("`edition` field not set in `[package]` section");
```

Reply:
Maybe we should also include the package id, WDYT?

```suggestion
                config.ui().warn(format!("`edition` field not set in `[package]` section for package {package_id}"));
```

Reply:
Name would be enough IMHO